opsdroid/core.py

```diff
@@ -101,6 +101,9 @@ def restart(self):
 
     def stop(self):
         """Stop the event loop."""
+        for connector in self.connectors:
```

Comment:
I assume it needs to be in this method somewhere, maybe even like this. This however isn't actually made to run, because it's cancelled right after?

Reply:
Yes. I would gather those disconnect tasks and wait for them to complete.

Reply:
How should I wait for them to complete? Using `run_until_complete` didn't work, I got something along the lines of "loop is already running."

Reply:
I think as you're already in a coroutine you can simply await the gather, you don't need to run the loop again.

Reply:
To be honest you don't even need to create them on the loop. Just wait them.

Change

```python
self.eventloop.create_task(connector.disconnect(self))`
```

to

```python
await connector.disconnect(self)
```

Reply:
Good point, perhaps then the `disconnect` method shouldn't be a coroutine.

Reply:
Certainly the use case I have for it, I would rather it be a coroutine, as the disconnect call is async like the rest of the lib. I could probably work around it though.

Reply:
Yep that's fair. Perhaps then the `stop` method should be a coroutine.